File: velocity/src/main/java/gg/norisk/velocity/api/NrcChannelRegister.java
package gg.norisk.velocity.api;

import org.slf4j.Logger;

import com.velocitypowered.api.proxy.ProxyServer;
import com.velocitypowered.api.proxy.messages.MinecraftChannelIdentifier;

import gg.norisk.core.common.Constants;

public class NrcChannelRegister {
    public static void register(ProxyServer server, Logger logger) {
        logger.info("Registering '" + Constants.NRC_CHANNEL + "' channel...");
        MinecraftChannelIdentifier channelIdentifier = MinecraftChannelIdentifier.from(Constants.NRC_CHANNEL);
        server.getChannelRegistrar().register(channelIdentifier);
        logger.info("Channel registration complete.");
    }
} 

File: velocity/src/main/kotlin/gg/norisk/velocity/api/NrcChannelRegister.kt
package gg.norisk.velocity.api

import com.velocitypowered.api.proxy.ProxyServer
import com.velocitypowered.api.proxy.messages.MinecraftChannelIdentifier
import gg.norisk.core.common.Constants.Companion.NRC_CHANNEL
import org.slf4j.Logger

object NrcChannelRegister {
    fun register(server: ProxyServer, logger: Logger) {
        logger.info("Registering '$NRC_CHANNEL' channel...")
        val channelIdentifier = MinecraftChannelIdentifier.from(NRC_CHANNEL)
        server.channelRegistrar.register(channelIdentifier)
        logger.info("Channel registration complete.")
    }
}
